### ISSUE TITLE
How can I specify the server hostname on the fly for the generated requests?

### ISSUE THREAD

**Issue by lavocatt** [7nohe/openapi-react-query-codegen#94]
Hello,

I'm looking to use `react-query` and the code gen for a project. I'd like to be able to specify on the fly in my React component the host that would receive the query. In my use case I have several machines that could potentially receive the queries I want to send.

This is my request for now:
```
  const { data, isError } = useQuery({
    queryKey: [useSecurityServiceLogin],
    queryFn: () => {
      return SecurityService.login({
        formData: {
          brokerName: getBrokerKey(broker, ordinal),
          userName: userName,
          password: password,
          jolokiaHost: jolokiaHost,
          port: jolokiaPort,
          scheme: protocol,
        },
      });
    },
  });

  return [!isError, data['jolokia-session-id']];
```
How can I put in ther the hostname and the port ?

Thanks.

**Comment by seriouslag** [comment]
@lavocatt 

Modifying the OpenAPI config sounds precisely like what you need.

Changing the `BASE` URL option on the OpenAPI config. 

Import the constant into your runtime and modify it before setting up the react app.

```typescript
/** main.tsx */
import { OpenAPI as OpenAPIConfig } from './openapi/requests/core/OpenAPI';
...
OpenAPIConfig.BASE = 'www.domain.com/api';
...
ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
    <QueryClientProvider client={queryClient}>
      <App />
    </QueryClientProvider>
  </React.StrictMode>
);

```

**Comment by lavocatt** [comment]
Thanks @seriouslag I have a question regarding this approach. It happens before  creating the `QueryClientProvider`. Is there any way to have this change more dynamic? For instance having a list of endpoints and sending a request to each of them ? Is this feasible?

**Comment by lavocatt** [comment]
Indeed, I can have a multiple set of machine running a server and querying each of them is one of my  use cases

**Comment by seriouslag** [comment]
Currently, there is no way to handle a per-request configuration. I opened an issue in the underlying service generator we use to add this feature.

Adding your use case there would be helpful to move this feature request along: https://github.com/hey-api/openapi-ts/issues/465
